protocol/x/subaccounts/keeper/isolated_subaccount.go

```diff
@@ -140,3 +138,193 @@ func isValidIsolatedPerpetualUpdates(
+		quoteQuantumsBeforeUpdate := new(big.Int).Sub(
+			updatedSubaccount.GetUsdcPosition(),
+			settledUpdateWithUpdatedSubaccount.AssetUpdates[0].GetBigQuantums(),
+		)
+		return &types.IsolatedPerpetualPositionStateTransition{
+			SubaccountId:  updatedSubaccount.Id,
+			PerpetualId:   perpetualUpdate.PerpetualId,
+			QuoteQuantums: quoteQuantumsBeforeUpdate,
+			Transition:    types.Opened,
+		}, nil
+	}
+
+	// The isolated perpetual position changed size but was not opened or closed.
+	return nil, nil
+}
+
+// transferCollateralForIsolatedPerpetual transfers collateral between an isolated collateral pool
+// and the cross-perpetual collateral pool based on whether an isolated perpetual position was
+// opened or closed in a subaccount.
+// Note: This uses the `x/bank` keeper and modifies `x/bank` state.
+func (k *Keeper) transferCollateralForIsolatedPerpetual(
+	ctx sdk.Context,
+	stateTransition *types.IsolatedPerpetualPositionStateTransition,
+) error {
+	// No collateral to transfer if no state transition.
+	if stateTransition == nil {
+		return nil
+	}
+
+	isolatedCollateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, stateTransition.PerpetualId)
+	if err != nil {
+		return err
+	}
+	var toModuleAddr sdk.AccAddress
+	var fromModuleAddr sdk.AccAddress
+
+	// If an isolated perpetual position was opened in the subaccount, then move collateral from the
+	// cross-perpetual collateral pool to the isolated perpetual collateral pool.
+	if stateTransition.Transition == types.Opened {
+		toModuleAddr = isolatedCollateralPoolAddr
+		fromModuleAddr = types.ModuleAddress
+		// If the isolated perpetual position was closed, then move collateral from the isolated
+		// perpetual collateral pool to the cross-perpetual collateral pool.
+	} else if stateTransition.Transition == types.Closed {
+		toModuleAddr = types.ModuleAddress
+		fromModuleAddr = isolatedCollateralPoolAddr
+	} else {
+		// Should never hit this.
+		return errorsmod.Wrapf(
+			types.ErrFailedToUpdateSubaccounts,
+			"Invalid state transition %v for isolated perpetual with id %d in subaccount with id %v",
+			stateTransition,
+			stateTransition.PerpetualId,
+			stateTransition.SubaccountId,
+		)
+	}
+
+	// If there are zero quantums to transfer, don't transfer collateral.
+	if stateTransition.QuoteQuantums.Sign() == 0 {
+		return nil
+	}
+
+	// Invalid to transfer negative quantums. This should already be caught by collateralization
+	// checks as well.
+	if stateTransition.QuoteQuantums.Sign() == -1 {
+		return errorsmod.Wrapf(
+			types.ErrFailedToUpdateSubaccounts,
+			"Subaccount with id %v %s perpteual position with perpetual id %d with negative collateral %s to transfer",
+			stateTransition.SubaccountId,
+			stateTransition.Transition.String(),
+			stateTransition.PerpetualId,
+			stateTransition.QuoteQuantums.String(),
+		)
+	}
+
+	// Transfer collateral between collateral pools.
+	_, coinToTransfer, err := k.assetsKeeper.ConvertAssetToCoin(
```

Comment:
informational q: what exactly are we converting for the collateral here?

Reply:
Converts between `quantums` (units that `x/subaccounts` use to track balances) and `coins` (units that `x/bank` use to track balances).